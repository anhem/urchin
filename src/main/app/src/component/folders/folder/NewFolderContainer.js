import React, {Component} from 'react'
import {connect} from 'react-redux'
import NewFolder from './NewFolder';
import {createFolder, setFolder} from '../../../action/folderAction';

export class NewFolderContainer extends Component {

    render() {
        return (
            <NewFolder folder={this.props.folder}
                       fieldErrors={this.props.fieldErrors}
                       callbacks={{
                           setFolder: this.props.setFolder,
                           createFolder: this.props.createFolder,
                       }}/>
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        setFolder: (folder) => {
            dispatch(setFolder(folder))
        },
        createFolder: (folder) => {
            dispatch(createFolder(folder))
        }
    }
};

const mapStateToProps = (state) => {
    return {
        folder: state.folderData.folder || {},
        fieldErrors: state.folderData.fieldErrors || {}
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(NewFolderContainer)