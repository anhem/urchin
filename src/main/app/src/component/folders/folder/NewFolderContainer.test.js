import React from 'react';
import {mount, shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import {NewFolderContainer} from "./NewFolderContainer";

describe('NewFolderContainer', () => {

    let props = {
        folder: {},
    };

    it('renders without crashing', () => {

        expect(mount(<NewFolderContainer {...props}/>).length).toEqual(1);
    });

    it('match snapshot', () => {
        expect(toJson(shallow(<NewFolderContainer {...props}/>))).toMatchSnapshot();
    });
});
