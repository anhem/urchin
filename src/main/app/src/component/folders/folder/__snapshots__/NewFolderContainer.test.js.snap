// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewFolderContainer match snapshot 1`] = `
<NewFolder
  callbacks={
    Object {
      "createFolder": undefined,
      "setFolder": undefined,
    }
  }
  folder={Object {}}
/>
`;
